grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -728,7 +729,46 @@ class QueryService implements Status {
         Collection result = Collections.emptyList()
 
         if (ids.size() > 0) {
-            result = Issue.findAllByIdInList(ids)
+            result = findIssuesSearchItems(ids)
+        }
+        result
+    }
+
+    /**
+     * Find issues and create DTOs to narrow the amount of data used in the search process
+     * @param issueIds
+     * @return
+     */
+    List<Issue> findIssuesSearchItems(ArrayList<Integer> issueIds) {
+        final String query = "SELECT i.id id, " +
+                "i.project_key projectKey, " +
+                "i.type type, " +
+                "i.status status, " +
+                "i.summary summary, " +
+                "i.reporter reporter, " +
+                "i.update_date updated, " +
+                "i.expiration_date expirationDate " +
+                "FROM issue i " +
+                "WHERE i.id IN (" + issueIds.join(",") + ")"
+
+        List<Issue> result = new ArrayList<Issue>()
+
+        getSqlConnection().rows(query).each{
+            Issue issueSearchItem = new Issue()
```

Comment:
Why create a new issue here? You could let hibernate give you a populated `Issue` object from the query which would avoid the awkward parsing, e.g. https://github.com/broadinstitute/orsp-pub/blob/feature/grails-app/services/org/broadinstitute/orsp/QueryService.groovy#L314

This is also a risk because it introduces a new, untested, pattern for hydrating objects. I do worry about what Hibernate will do with all of these dirty (unsaved) new objects and whether that introduces a separate set of performance/memory concerns.

Reply:
If I do as the example the search performance drops drastically from seconds up to five minutes. On the other hand we're not doing anything with the Issue objects but showing the results and only the fields that are mapped, there's no need of other info. So we're not using unnecessary memory.

Reply:
The concern isn't that we are doing anything with unnecessary memory, the concern is that Grails/Hibernate is doing something. `Issue` is a domain object which means Grails treats it differently than other classes. It could be the case that we're creating memory leaks this way, but I just don't know for sure which is why I bring up the question.

Another alternative is to create some other kind of object that is not a domain object and pass that along as the list of things to return as search results.

Reply:
I don't quite get the concern. The issue objects are read only in this scenario. Another thing, I already try to use a DTO and got very complex very fast. But when I added the index I found that I there was no need to use a DTO because it was very fast and got a lot simpler. Also I found that I was going to have to change the way is used `PermissionService.issueIsForbiden`
In this commit is more or less that approach 87cba6a

Reply:
This is also one of my concerns: 

> The issue objects are read only in this scenario.

This method assumes that these are read-only, but there is nothing to guarantee that. Imagine if some controller called this method, got a bunch of issues back, and tried to do stuff with them. You'd now have modified `Issue` objects that could be re-saved, and only partially (meaning data will be deleted) because you didn't fully populate them from this method. You've chosen a non-standard way to use Grails domain classes which could lead to real problems.